### PR TITLE
Style: Link method argument definitions

### LINKED ISSUE #574: Consider alternate styling/linking for method argument definitions
Method arguments are currently detailed with this markdown (using compute() as an example):

```
<div>
    **Arguments:**
      - *graph*: an {{MLGraph}}. The compiled graph to be executed.
      - *inputs*: an {{MLNamedArrayBufferViews}}. The resources of inputs. Will be [=MLNamedArrayBufferViews/transfer|transferred=] if there are no validation errors.
      - *outputs*: an {{MLNamedArrayBufferViews}}. The pre-allocated resources of required outputs. Will be [=MLNamedArrayBufferViews/transfer|transferred=] if there are no validation errors.
</div>
```
which renders as:
<kbd></kbd>

There are a few other forms used in other specs worth considering to improve maintainability.

## Alternative 1: Make args into definitions

A minimal change makes the argument names become definitions which are get automatically linked, to help catch errors.
```
<div dfn-for="MLContext/compute(graph, inputs, outputs)" dfn-type="argument">
    **Arguments:**
      - <dfn>graph</dfn>: an {{MLGraph}}. The compiled graph to be executed.
      - <dfn>inputs</dfn>: an {{MLNamedArrayBufferViews}}. The resources of inputs. Will be [=MLNamedArrayBufferViews/transfer|transferred=] if there are no validation errors.
      - <dfn>outputs</dfn>: an {{MLNamedArrayBufferViews}}. The pre-allocated resources of required outputs. Will be [=MLNamedArrayBufferViews/transfer|transferred=] if there are no validation errors.
</div>
```
which renders as:
<kbd></kbd>

## Alternative 2: Style as definition list

Extending the above using a definition list, similar to what is currently done for dictionaries:
```
**Arguments:**
<dl dfn-for="MLContext/compute(graph, inputs, outputs)" dfn-type="argument">

: <dfn>graph</dfn>
:: an {{MLGraph}}. The compiled graph to be executed.

: <dfn>inputs</dfn>
:: an {{MLNamedArrayBufferViews}}. The resources of inputs. Will be [=MLNamedArrayBufferViews/transfer|transferred=] if there are no validation errors.

: <dfn>outputs</dfn>
:: an {{MLNamedArrayBufferViews}}. The pre-allocated resources of required outputs. Will be [=MLNamedArrayBufferViews/transfer|transferred=] if there are no validation errors.

</dl>
```
which renders as:
<kbd></kbd>

## Alternative 3: Auto-generated table

This is a much more radical departure from what the spec has today. The spec source is extremely compact, which eases maintenance. Argument type, nullability, and optionality are filled in by Bikeshed automatically, reducing errors. On the flip side, I personally find the big table ugly.

```
<pre class=argumentdef for="MLContext/compute(graph, inputs, outputs)">
graph: The compiled graph to be executed.
inputs: The resources of inputs. Will be [=MLNamedArrayBufferViews/transfer|transferred=] if there are no validation errors.
outputs: The pre-allocated resources of required outputs. Will be [=MLNamedArrayBufferViews/transfer|transferred=] if there are no validation errors.
</pre>
```
which renders as:
<kbd></kbd>

------

Note that for all of these the return value of the method is not handled.

### PR DESCRIPTION
**Pull request by inexorabletash**
Also adds a lint rule to enforce that every argument given in IDL is linked to a prose definition for it, and a lint rule to look for argument definitions that don't match their methods.

Fixes #574

***
<a href="https://pr-preview.s3.amazonaws.com/inexorabletash/webnn/pull/721.html" title="Last updated on Jul 12, 2024, 6:46 PM UTC (c2001d9)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/721/af67a72...inexorabletash:c2001d9.html" title="Last updated on Jul 12, 2024, 6:46 PM UTC (c2001d9)">Diff</a>

### CODE REVIEW

**Review by huningxin — APPROVED** [review]
LGTM!

File: index.bs [hunk 1]
```diff
@@ -5976,7 +5976,7 @@ partial interface MLGraphBuilder {
 };
 </script>
 
-<div dfn-for="MLGraphBuilder/where(condition, input, other)" dfn-type=argument>
+<div dfn-for="MLGraphBuilder/where(condition, trueValue, falseValue)" dfn-type=argument>
     **Arguments:**
         - <dfn>condition</dfn>: an {{MLOperand}}. The condition tensor.
         - <dfn>trueValue</dfn>: an {{MLOperand}}. The trueValue tensor from which the value is selected when the condition of the corresponding element is set to true.
```
Comment (inexorabletash) [inline index.bs:5982]: Aside: repeating `trueValue` here is a little odd. How about just "The tensor from which..." ?

Comment (fdwr) [reply index.bs:5982]: Fine with me.

Comment (inexorabletash) [reply index.bs]: Done in c2001d9479b803ea87837e49cf6e790af9d24552

**Review by fdwr — APPROVED** [review]
👍 TY JB.

**Pull request merged**